ndcube/ndcube.py

```diff
@@ -952,3 +962,234 @@ def to(self, new_unit, **kwargs):
+                           [0, 0, 1, 0, 0, 0]])
+
+        and you want to sum over 2x2 non-overlapping sub-arrays.  This summing can
+        be done by reshaping the array::
+
+             y = x.reshape(3,2,3,2)
+
+        and then summing over the 1st and third directions::
+
+             y2 = y.sum(axis=3).sum(axis=1)
+
+        which gives the expected array::
+
+             array([[0, 3, 2],
+                    [2, 0, 4],
+                    [1, 2, 2]])
+
+        **Defining Custom Error Propagation**
+        To perform custom uncertainty propagation, a function must be provided via the
+        propgate_uncertainty kwarg. This function must accept, although doesn't have to
+        use, the following args:
+
+        uncertainty: `astropy.nddata.NDUncertainty` but not `astropy.nddata.UnknownUncertainty`
+            The uncertainties associated with the data.
+        data: array-like
+            The data associated with the above uncertainties.
+            Must have same shape as uncertainty.
+        mask: array-like of `bool` or `None`
+            Indicates whether any uncertainty elements should be ignored in propagation.
+            True elements cause corresponding uncertainty elements to be ignored.
+            False elements cause corresponding uncertainty elements to be propagated.
+            Must have same shape as above.
+            If None, no uncertainties are ignored.
+
+        All kwarg inputs to the rebin method are also passed on transparently to the
+        propagation function. Hence additional inputs to the propagation function can be
+        included as kwargs to `ndcube.NDCube.rebin`.
+
+        The shape of the uncertainty, data and mask inputs are such that the first
+        dimension represents the pixels in a given bin whose data and uncertainties
+        are aggregated by the rebin process.  The shape of the remaining dimensions
+        must be the same as the final rebinned data.  A silly but informative
+        example of a custom propagation function might be::
+
+             def my_propagate(uncertainty, data, mask, **kwargs):
+                 # As a silly example, propagate uncertainties by summing those in same bin.
+                 # Note not all args are used, but function must accept them.
+                 n_pixels_per_bin = data.shape[0]  # 1st dimension of inputs gives pixels in bin.
+                 final_shape = data.shape[1:]  # Trailing dims give shape of put rebinned data.
+                 # Propagate uncerts by adding them.
+                 new_uncert = numpy.zeros(final_shape)
+                 for i in range(n_pixels_per_bin):
+                     new_uncert += uncertainty.array[i]
+                 # Alternatively: new_uncerts = uncertainty.array.sum(axis=0)
+                 return type(uncertainty)(new_uncert)  # Convert to original uncert type and return.
+        """
+        # Sanitize input.
+        if new_unit is None:
+            new_unit = self.unit
```

Comment:
```suggestion
        new_unit = new_unit or self.unit
```

Reply:
Won't this return a boolean?  We want it be a unit.